lib/resty/kafka/basic-consumer.lua

```diff
@@ -0,0 +1,143 @@
+local client = require("resty.kafka.client")
+local broker = require("resty.kafka.broker")
+local protocol_consumer = require("resty.kafka.protocol.consumer")
+local Errors = require("resty.kafka.errors")
+
+local ngx_log = ngx.log
+local INFO = ngx.INFO
+
+
+local _M = {}
+local mt = { __index = _M }
+
+function _M.new(self, broker_list, client_config)
+    local opts = client_config or {}
+
+    local cli = client:new(broker_list, client_config)
+    local p = setmetatable({
+        client = cli,
+        correlation_id = 1,
+        isolation_level = opts.isolation_level or 0,
+        client_rack = opts.client_rack or "default",
+        socket_config = cli.socket_config,
+    }, mt)
+
+    return p
+end
+
+
+--- Get the available offsets of the partition of the specified topic
+-- When the error in request, offset will be nil and err will be the error message.
+-- @author bzp2010 <bzp2010@apache.org>
+-- @param self
+-- @param topic      The name of topic
+-- @param partition  The partition of topic
+-- @param timestamp  The starting timestamp of the obtained message offset
+-- @return offset    The obtained offset value, which is in a table, may be nil
+-- @return err       The error of request, may be nil
+function _M.list_offset(self, topic, partition, timestamp)
+    timestamp = timestamp or protocol_consumer.LIST_OFFSET_TIMESTAMP_FIRST
+
+    local cli = self.client
+    local broker_conf = cli:choose_broker(topic, partition)
+    local bk, err = broker:new(broker_conf.host, broker_conf.port, self.socket_config, broker_conf.sasl_config)
+    if not bk then
+        return nil, err
+    end
+
+    local req, err = protocol_consumer.list_offset_encode(self, {
+        topic_num = 1,
+        topics = {
+            [topic] = {
+                partition_num = 1,
+                partitions = {
+                    [partition] = {
+                        timestamp = timestamp
+                    }
+                },
+            }
+        },
+    })
+    if not req then
+        return nil, err
+    end
+
+    local resp, err = bk:send_receive(req)
+    if not resp then
+        return nil, err
+    end
+
+    local result = protocol_consumer.list_offset_decode(resp)
+    local data = result.topics[topic].partitions[partition]
```

Comment:
what if the topic is not existing? please add some tests to cover it.

Reply:
After testing, I found that when the topic does not exist, an operate will automatically create it. 🤔

Reply:
fixed, The `not found topic` and `not found partition` cases has been added.